WordPress/Classes/Utility/WPWebViewController.m

```diff
@@ -30,123 +64,43 @@ - (void)dealloc
-    }
-
-    [self setLoading:NO];
-    self.backButton.enabled = NO;
-    self.forwardButton.enabled = NO;
-    self.backButton.accessibilityLabel = NSLocalizedString(@"Back", @"Spoken accessibility label");
-    self.forwardButton.accessibilityLabel = NSLocalizedString(@"Forward", @"Spoken accessibility label");
-    self.refreshButton.accessibilityLabel = NSLocalizedString(@"Refresh", @"Spoken accessibility label");
-
-    if (IS_IPHONE) {
-        if (!self.hidesLinkOptions) {
-            [WPStyleGuide setRightBarButtonItemWithCorrectSpacing:self.optionsButton forNavigationItem:self.navigationItem];
-        }
-    } else {
-        // We want the refresh button to be borderless, but buttons in navbars want a border.
-        // We need to compose the refresh button as a UIButton that is used as the UIBarButtonItem's custom view.
-        UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
-        [btn setImage:[UIImage imageNamed:@"sync_lite"] forState:UIControlStateNormal];
-        [btn setImage:[UIImage imageNamed:@"sync"] forState:UIControlStateHighlighted];
-
-        btn.frame = CGRectMake(0.0f, 0.0f, 30.0f, 30.0f);
-        btn.autoresizingMask =  UIViewAutoresizingFlexibleHeight;
-        [btn addTarget:self action:@selector(reload) forControlEvents:UIControlEventTouchUpInside];
-        self.refreshButton.customView = btn;
-
-        if (self.navigationController && self.navigationController.navigationBarHidden == NO) {
-            CGRect frame = self.webView.frame;
-            frame.origin.y -= self.iPadNavBar.frame.size.height;
-            frame.size.height += self.iPadNavBar.frame.size.height;
-            self.webView.frame = frame;
-            self.navigationItem.rightBarButtonItem = self.refreshButton;
-            self.title = NSLocalizedString(@"Loading...", @"");
-            [self.iPadNavBar removeFromSuperview];
-            self.iPadNavBar = self.navigationController.navigationBar;
-        } else {
-            self.refreshButton.customView = btn;
-            self.iPadNavBar.topItem.title = NSLocalizedString(@"Loading...", @"");
-        }
-        self.loadingLabel.text = NSLocalizedString(@"Loading...", @"");
-    }
-
-    self.toolbar.translucent = NO;
-    self.toolbar.barTintColor = [WPStyleGuide littleEddieGrey];
-    self.toolbar.tintColor = [UIColor whiteColor];
-
-    self.optionsButton.enabled = NO;
-    self.webView.scalesPageToFit = YES;
-    self.scrollView.decelerationRate = UIScrollViewDecelerationRateNormal;
-
-    if (self.url) {
-        [self refreshWebView];
-    } else {
-        self.navigationItem.title = NSLocalizedString(@"Loading...", @"");
-        [self.webView loadHTMLString:self.detailHTML baseURL:[NSURL URLWithString:@"https://en.wordpress.com"]];
-    }
-}
+    NSAssert(self.webView,          @"Missing Outlet!");
+    NSAssert(self.progressView,     @"Missing Outlet!");
+    NSAssert(self.optionsButton,    @"Missing Outlet!");
```

Comment:
Nice touch.

As a general rule when asserting properties I recommend asserting the ivars directly.  Even though right now the getters don't do any crazy stuff (like lazy initialization), they may in the future - thus possibly breaking here in a non-obvious way.